glue/core/data_factories/fits.py

```diff
@@ -114,14 +114,16 @@ def new_data(suffix=True):
             if is_image_hdu(hdu):
                 shape = hdu.data.shape
                 coords = coordinates_from_header(hdu.header)
+                units = hdu.header["BUNIT"] if "BUNIT" in hdu.header else None
```

Comment:
I think you can write this as ``hdu.header.get('BUNIT')`` as that will return ``None`` if the key doesn't exist.